streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -107,9 +107,6 @@ abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable
    */
   def onStop()
 
-  /** Override this to specify a preferred location (hostname). */
-  def preferredLocation : Option[String] = None
```

Comment:
Why move this function from `Receiver` to `ReceiverInputDStream`? I think for people who customized `Receiver` and override this function may fail to compile. Is there any specific reason?